mpi-proxy-split/seq_num.cpp

```diff
@@ -170,16 +171,19 @@ void commit_begin(MPI_Comm comm) {
     // Stop before the critical section during checkpoint time.
     if (ckpt_pending && check_seq_nums()) {
       current_phase = STOP_BEFORE_CS;
-      while (!freepass && ckpt_pending);
-      freepass = false;
+      while(ckpt_pending){
```

Comment:
QUESTION:  Isn't this the same as:
```
if (ckpt_pending) {
  sem_wait(&freepass_sem));
}
```
(And we could check the return value of sem_wait() if we're worried about that.)

Reply:
It's different, because of ckpt_pending becomes false after entering the loop then we will leave the loop, but if ckpt_pending becomes false after entering the if statement we would not exit then

Reply:
@chirag-singh-memverge We can exit the loop at resume/restart time with a sem_trywait() and a sem_post().